lib/releng_common/releng_common/auth.py

```diff
@@ -60,163 +60,158 @@ def __init__(self, credentials):
-
-        expanded = []
-
-        def _expand(scopes):
-            for s in scopes:
-                if s in expanded or not s.startswith('assume:'):
-                    continue
-
-                # Skip useless roles
-                if s.startswith('assume:assume:') \
-                    or s.startswith('assume:hook-id:') \
-                    or s in (
-                        'assume:project:taskcluster:tutorial',
-                    ):
-                    continue
-
-                # Solve assume: role
-                try:
-                    role = auth.role(s)
-                    expanded.append(s)
-                    scopes += _expand(role.get('scopes', []))
-                    scopes += _expand(role.get('expandedScopes', []))
-                except Exception as e:
-                    logger.warning('Failed to expand role {} : {}'.format(s, e))
-                    expanded.append(s)
-
-            return scopes
-
-        return self.scopes # should work without expand now
-        return _expand(self.scopes)
-
-    def taskcluster_secrets(self):
+        return self.credentials['scopes']
+
+    # XXX: this should not be here
+
+    def taskcluster_options(self):
         """
         Configure the TaskCluster Secrets client
         with optional target HAWK header
         """
         target_header = request.environ.get('HTTP_X_AUTHORIZATION_TARGET')
         if not target_header:
             raise Exception('Missing X-AUTHORIZATION-TARGET header')
-
-        options = {
+        return {
             'credentials' : {
                 'hawkHeader' : target_header
             }
         }
-        return taskcluster.Secrets(options)
 
     def get_secret(self, name):
         """
         Helper to read a Taskcluster secret
         """
-        secrets = self.taskcluster_secrets()
+        secrets = taskcluster.Secrets(self.taskcluster_options())
         secret = secrets.get(name)
         if not secret:
             raise Exception('Missing TC secret {}'.format(name))
         return secret['secret']
 
+
 class Auth(object):
 
-    def __init__(self, login_manager):
-        self.login_manager = login_manager
+    def __init__(self, anonymous_user):
+        self.login_manager = LoginManager()
+        self.login_manager.anonymous_user = anonymous_user
+        self.app = None
 
-    def require_scope(self, scope):
-        self.require_login()
-        import pdb
-        pdb.set_trace()
+    def init_app(self, app):
+        self.app = app
+        self.login_manager.init_app(app)
 
-    def require_login(self):
-        import pdb
-        pdb.set_trace()
 
-def scopes_required(scopes):
-    """
-    Decorator for Flask views to require
-    some Taskcluster scopes for the current user
-    """
-    assert isinstance(scopes, list)
+    def _require_scopes(self, scopes):
+        response = self._require_login()
+        if request.path is not None:
```

Comment:
it should be:
```python
if response is not None:
  return response
```

because you wan to return the abort() from _require_login if the check fails.